modelbuilders/xgb_mb.py

```diff
@@ -0,0 +1,153 @@
+    'scale_pos_weight': params.scale_pos_weight,
+    'grow_policy': params.grow_policy,
+    'max_leaves': params.max_leaves,
+    'max_bin': params.max_bin,
+    'objective': params.objective,
+    'seed': params.seed,
+    'single_precision_histogram': params.single_precision_histogram,
+    'enable_experimental_json_serialization': params.enable_experimental_json_serialization
+}
+
+if params.threads != -1:
+    xgb_params.update({'nthread': params.threads})
+
+if 'OMP_NUM_THREADS' in environ.keys():
+    xgb_params['nthread'] = int(environ['OMP_NUM_THREADS'])
+
+columns: Tuple[str, ...] = ('batch', 'arch', 'prefix', 'function', 'threads', 'dtype', 'size', 'num_trees')
+
+if params.objective.startswith('reg'):
+    task = 'regression'
+    metric_name, metric_func = 'rmse', rmse_score
+    columns += ('rmse', 'time')
+else:
+    task = 'classification'
+    metric_name, metric_func = 'accuracy[%]', get_accuracy
+    columns += ('n_classes', 'accuracy', 'time')
+    if 'cudf' in str(type(y_train)):
+        params.n_classes = y_train[y_train.columns[0]].nunique()
+    else:
+        params.n_classes = len(np.unique(y_train))
+    if params.n_classes > 2:
+        xgb_params['num_class'] = params.n_classes
+
+t_creat_train, dtrain = measure_function_time(xgb.DMatrix, X_train, params=params, label=y_train)
+
+t_creat_test, dtest = measure_function_time(xgb.DMatrix, X_test, params=params)
+
+def fit(dmatrix=None):
+    if dmatrix is None:
+        dmatrix = xgb.DMatrix(X_train, y_train)
+    return xgb.train(xgb_params, dmatrix, params.n_estimators)
+
+def predict():
+    dmatrix = xgb.DMatrix(X_test, y_test)
+    return model_xgb.predict(dmatrix)
+
+t_train, model_xgb = measure_function_time(fit, None if params.count_dmatrix else dtrain, params=params)
+y_train_pred = model_xgb.predict(dtrain)
+train_metric = metric_func(y_train, y_train_pred)
+
+t_xgb_pred, y_test_pred = measure_function_time(predict, params=params)
+test_metric_xgb = metric_func(y_test, y_test_pred)
+
+t_trans, model_daal = measure_function_time(daal4py.get_gbt_model_from_xgboost, model_xgb, params=params)
+
+if hasattr(params, 'n_classes'):
+    predict_algo = daal4py.gbt_classification_prediction(nClasses=params.n_classes, 
+        resultsToEvaluate='computeClassLabels', fptype='float')
+    t_daal_pred, daal_pred = measure_function_time(predict_algo.compute, X_test, model_daal, params=params)
+    test_metric_daal = metric_func(y_test, daal_pred.prediction)
+else:
+    predict_algo = daal4py.gbt_regression_prediction()
+    t_daal_pred, daal_pred = measure_function_time(predict_algo.compute, X_test, model_daal, params=params)
+    test_metric_daal = metric_func(y_test, daal_pred.prediction)
+
+print_output(library='modelbuilders', algorithm=f'xgboost_{task}_and_modelbuilder',
+             stages=['xgb_train_dmatrix_create', 'xgb_test_dmatrix_create', 'xgb_training', 'xgb_prediction', 
+                'xgb_to_daal_conv', 'daal_prediction'],
```

Comment:
Use flake8 to correct formatting:
`pip/conda install flake8`
`flake8 <folder or file names>`
You can ignore 'too long line' if line is complicated

Reply:
done it for every file, which I added / changed

Reply:
I also used autopep8 formatting, so every line is less or equal 100 characters now